internal/docker/docker.go

```diff
@@ -126,12 +132,31 @@ func (dc *dockerClient) LoadContainerList(ctx context.Context) error {
 	return nil
 }
 
-// FetchContainerStatsAndConvertToMetrics will query the desired container stats and send
-// converted metrics to the results channel, since this is intended to be run in a goroutine.
-func (dc *dockerClient) FetchContainerStatsAndConvertToMetrics(
+// FetchContainerStatsAsJSON will query the desired container stats
+// and return them as StatsJSON
+func (dc *Client) FetchContainerStatsAsJSON(
+	ctx context.Context,
+	container Container,
+) (*dtypes.StatsJSON, error) {
+	containerStats, err := dc.FetchContainerStats(ctx, container)
+	if err != nil {
+		return nil, err
+	}
+
+	statsJSON, err := dc.toStatsJSON(containerStats, &container)
```

Comment:
Can the rest of this method just be
```
	return dc.toStatsJSON(containerStats, &container)
```

Reply:
Ah yes, I had changed this from a pointer to just the direct struct and missed this, thanks.